### PR TITLE
adjusting pot create command to take into consideration whether a jai…

### PR DESCRIPTION
**Pull request by bretton**
_This is untested, adapting from my own build scripts._

When flavour.sh has `RUNS_IN_NOMAD=false`, then we don't include the flavour+4 file in the `pot create` command.

We only want to include the flavour+4 file when flavour.sh has `RUNS_IN_NOMAD=true`

### CODE REVIEW

**Comment by grembo** [comment]
Hi @bretton, thanks for your contribution.

What is the exact purpose of the change?

Is this supposed for flavours that support both running in Nomad and not running in Nomad - so users would configure what they want by modifying the flavour shell script?
Or is it more than flavours are designed to do one or the other (not changed by users)?
Or is it potentially both?

If it's supposed to be user configurable, I might make sense to add a command line switch to potman. If not, we could simply check for the existence of "$FLAVOUR"+4 to decide if we want to pass it in or not.

Thanks!

**Comment by bretton** [comment]
> What is the exact purpose of the change?

Hi @grembo

Purpose is to correct a build error that may occur, where the `flavour+4` is included in a pot jail. The resulting pot jail image won't handle reboots. 

I also considered simply removing the `flavour+4` file, but the `pot create` command in still has to be edited to exclude it. 

It then made more sense to check the for the variable `RUNS_IN_NOMAD`, and have the `pot build` step in custom tooling handle the decision to exclude or include the `flavour+4` file automatically.

This way a template `flavour` directory could always have the necessary files to handle both types of image, and the status of `RUNS_IN_NOMAD` determines the resulting image type.

File: potbuild.sh [hunk 1]
```diff
@@ -65,6 +65,14 @@ for file in $FLAVOUR_FILES; do
   fi
 done
 
+# searching for jail-based pot images
+NOMADCHECK=$(grep "^RUNS_IN_NOMAD=false" "$FLAVOURS_DIR"/"$FLAVOUR"/"$FLAVOUR".sh)
```
Comment (grembo) [inline potbuild.sh:69]: It would be better to check the return value of grep here (returns 1 on match).

Comment (grembo) [reply potbuild.sh:69]: Quoting $FLAVOUR isn't necessary, as it's guaranteed to contain nothing strange.

File: potbuild.sh [hunk 2]
```diff
@@ -157,7 +172,7 @@ mv \
   _build/artifacts/.
 
 # if DEBUG is enabled, dump the variables
-if [ $DEBUG -eq 1 ]; then
+if [ "$DEBUG" -eq 1 ]; then
```
Comment (grembo) [inline potbuild.sh:175]: Technically not necessary (DEBUG is normalized), but still best practice :+1:

**Review by grembo — COMMENTED** [review]
I get the point, thank you!

After some discussion, I think it might be better to approach this in a broader sense by adding a configuration file to the flavour that contains this piece of information.

**Pull request closed without merging**